### PR TITLE
PPF-477-Add polling to check for key visibility upgrade

### PR DESCRIPTION
**Pull request by vhande**
### Added
- polling to check for key visibility upgrade

---
Ticket: https://jira.uitdatabank.be/browse/PPF-477

### CODE REVIEW

File: resources/ts/types/Integration.ts [hunk 1]
```diff
@@ -32,4 +37,5 @@ export type Integration = {
   authClients: AuthClient[];
   legacyAuthConsumers: LegacyAuthConsumer[];
   keyVisibility: KeyVisibility;
+  keyVisibilityUpgrade: KeyVisibilityUpgrade;
```
Comment (simon-debruijn) [inline resources/ts/types/Integration.ts:40]:
```suggestion
  keyVisibilityUpgrade: KeyVisibilityUpgrade | null;
```

This can also be null no?

**Review by simon-debruijn — CHANGES_REQUESTED** [review]

File: resources/ts/types/Integration.ts (hunk 1 above)
Comment (LucWollants) [reply resources/ts/types/Integration.ts:40]: This is indeed not applicable to all integrations, only the integrations for which v2 keys are requested.

File: resources/ts/Components/Integrations/Detail/CredentialsAuthClients.tsx [hunk 2]
```diff
@@ -63,26 +70,31 @@ export const CredentialsAuthClients = ({
...
-              i18nKey="details.credentials.uitid_alert"
-              components={[
-                <Link
-                  key={t("details.credentials.uitid_alert")}
-                  href={t("details.credentials.uitid_alert_link")}
-                  className="text-publiq-blue-dark hover:underline mb-3"
-                />,
-              ]}
-            />
+        keyVisibility === KeyVisibility.v1 && !!keyVisibilityUpgrade ? (
```
Comment (Anahkiasen) [inline resources/ts/Components/Integrations/Detail/CredentialsAuthClients.tsx:73]: The first part of the condition is already covered by the ternary but more generally I would avoid the nested ternary as I feel it's hard to parse mentally. Could we extract the subcondition to a variable perhaps?

File: resources/ts/Components/Integrations/Detail/Credentials.tsx [hunk 3]
```diff
@@ -40,7 +41,11 @@ export const Credentials = ({
   const hasAnyCredentials = Boolean(
     legacyAuthConsumers.length || authClients.length
   );
-  usePolling(!hasAnyCredentials, { only: ["integration"] });
+  usePolling(
+    !hasAnyCredentials ||
+      (keyVisibility === KeyVisibility.v1 && !!keyVisibilityUpgrade),
```
Comment (Anahkiasen) [inline resources/ts/Components/Integrations/Detail/Credentials.tsx:46]: Minor but I would maybe but the second part of the condition in a variable like `hasAnyCredentials` was for readability purposes

**Review by Anahkiasen — CHANGES_REQUESTED** [review]
Mostly code readability comments but functionality wise all good 👍

**Review by simon-debruijn — APPROVED** [review]

File: resources/ts/Components/Integrations/Detail/CredentialsAuthClients.tsx [hunk 4]
```diff
@@ -62,27 +70,32 @@ export const CredentialsAuthClients = ({
...
-              components={[
-                <Link
-                  key={t("details.credentials.uitid_alert")}
-                  href={t("details.credentials.uitid_alert_link")}
-                  className="text-publiq-blue-dark hover:underline mb-3"
-                />,
-              ]}
-            />
+      {isKeyVisibilityV1 ? (
+        isKeyVisibilityV1 && !!keyVisibilityUpgrade ? (
```
Comment (Anahkiasen) [inline resources/ts/Components/Integrations/Detail/CredentialsAuthClients.tsx:74]: My previous comment still stands here since we check twice for `isKeyVisibilityV1`, since it cannot check the second condition without checking the first. But since it's a plain boolean check it's not costly at runtime so not really major, so good for me either way

**Review by Anahkiasen — APPROVED** [review]
